index.js

```diff
@@ -221,3 +221,84 @@ if (crypto) {
     activator
   );
 }
+
+var instrumentPromise = !!global.Promise;
+
+// Check that global Promise is native
+if (instrumentPromise) {
+  // shoult not use any methods that have already been wrapped
+  var promiseListener = process.addAsyncListener({
+    create: function create() {
```

Comment:
that is exactly what this tests. if they use an async method, this create will be called synchronously, and tell us that it is not the native promise implementation.